statsd/telemetry.go

```diff
@@ -113,39 +113,45 @@ type Telemetry struct {
 }
 
 type telemetryClient struct {
-	c          *Client
-	tags       []string
-	aggEnabled bool // is aggregation enabled and should we sent aggregation telemetry.
-	tagsByType map[metricType][]string
-	sender     *sender
-	worker     *worker
-	lastSample Telemetry // The previous sample of telemetry sent
+	sync.RWMutex // used mostly to change the transport tag.
+
+	c                 *Client
+	aggEnabled        bool // is aggregation enabled and should we sent aggregation telemetry.
+	transport         string
+	tags              []string
+	tagsByType        map[metricType][]string
+	transportTagKnown bool
+	sender            *sender
+	worker            *worker
+	lastSample        Telemetry // The previous sample of telemetry sent
 }
 
-func newTelemetryClient(c *Client, transport string, aggregationEnabled bool) *telemetryClient {
+func clientTransportTag(transport string) string {
+	if transport == writerNameUDS || transport == writerNameUDSDatagram {
+		return writerNameUDS // For backward compatibility.
+	}
+	return transport
```

Comment:
This function seems redundant now, since we return correct transport tag directly from the client?

Reply:
removed